shardy/dialect/sdy/ir/attrs.td

```diff
@@ -73,6 +73,11 @@ def Sdy_Mesh : AttrDef<Sdy_Dialect, "Mesh"> {
     - A mesh with two axes and implicit device IDs iota(6): <["a"=2, "b"=3]>
     - A mesh with two axes and explicit device IDs specifying the device
       ordering: <["a"=3, "b"=2], device_ids=[0, 2, 4, 1, 3, 5]>
+
+    **Constraints:**
+    - Elements in `axes` must not have duplicate names.
+    - The product of axis sizes must match the number of devices.
+    - If `device_ids` is specified, all of its elements must be non-negative.
```

Comment:
We are also missing:
- sorted `device_ids` must be `iota(product(axis_sizes))`
- `device_ids` shouldn't be equal to `iota(product(axis_sizes))`